scripts/o3de/o3de/android_support.py

```diff
@@ -0,0 +1,2166 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+import argparse
+import configparser
+import datetime
+import fnmatch
+import imghdr
+import json
+import logging
+import os
+import re
+import platform
+import shutil
+import stat
+import string
+import subprocess
+
+from enum import Enum
+from o3de import command_utils, manifest, utils
+from packaging.version import Version
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+logging.basicConfig(format=utils.LOG_FORMAT)
+logger = logging.getLogger('o3de.android')
+
+DEFAULT_READ_ENCODING = 'utf-8'    # The default encoding to use when reading from a text file or stream
+DEFAULT_WRITE_ENCODING = 'ascii'
+ENCODING_ERROR_HANDLINGS = 'ignore'     # What to do if we encounter any encoding errors
+
+# Set platform specific values and extensions
+if platform.system() == 'Windows':
+    EXE_EXTENSION = '.exe'
+    O3DE_SCRIPT_EXTENSION = '.bat'
+    SDKMANAGER_EXTENSION = '.bat'
+    GRADLE_EXTENSION = '.bat'
```

Comment:
just FYI I had a tricky situation come up on Windows when using choco to install gradle using `Cmder` which emulated bash, it did not have the `.bat` version for some reason and I ended up adding the ability to specify the gradle tool path https://github.com/o3de/o3de/compare/development...aws-lumberyard-dev:o3de:AMZN-alexpete/upgrade-gradle#diff-4df5f0f2791e5d71db4f914953168f4b9092669d86c5321b48bff4a72f016dc7R49